Missing generated/16px/paths in @blueprints/icons

<!-- IF YOU ARE A PALANTIR EMPLOYEE, DO NOT POST INTERNAL LINKS OR REFERENCES HERE -->

#### Environment

- __Package version(s)__: @blueprints/icons v4.6.1
- __Operating System__: N/A
- __Browser name and version__: N/A

#### Actual behavior

CI for depedabot updates is failing due to missing files in the @blueprints/icons v4.6.1 package.

Example error message and stack trace from https://github.com/pybricks/pybricks-code/actions/runs/3191502668/jobs/5207890478#step:6:3181

```
    Cannot find module './generated/16px/paths' from 'node_modules/@blueprintjs/icons/lib/cjs/iconSvgPaths.js'

    Require stack:
      node_modules/@blueprintjs/icons/lib/cjs/iconSvgPaths.js
      node_modules/@blueprintjs/icons/lib/cjs/index.js
      node_modules/@blueprintjs/core/lib/cjs/components/icon/icon.js
      node_modules/@blueprintjs/core/lib/cjs/components/button/abstractButton.js
      node_modules/@blueprintjs/core/lib/cjs/components/button/buttons.js
      node_modules/@blueprintjs/core/lib/cjs/components/alert/alert.js
      node_modules/@blueprintjs/core/lib/cjs/components/index.js
      node_modules/@blueprintjs/core/lib/cjs/index.js
      test/index.tsx
      src/notifications/i18n.test.ts
```

Checking the `node_modules` folder on a local install, we can see that indeed there is no `generated/` folder. So it seems like this folder was not included in the build that was published to npm for some reason.

#### Expected behavior

New packages should not be broken :wink: 

#### Possible solution

<!-- if you have any ideas -->

Similar issue while we build the application after getting above mentioned version of bp:
Failed to compile.
Module not found: Error: Can't resolve './generated/16px/paths' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconCodepoints.js 16:0-84
Module not found: Error: Can't resolve './generated/16px/blueprint-icons-16' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconNames.js 20:0-72
Module not found: Error: Can't resolve './generated/16px/blueprint-icons-16' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconSvgPaths.js 17:0-57
Module not found: Error: Can't resolve './generated/16px/paths' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconSvgPaths.js 18:0-57
Module not found: Error: Can't resolve './generated/20px/paths' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/index.js 16:0-90
export 'IconSvgPaths16' (reexported as 'IconSvgPaths16') was not found in './iconSvgPaths' (possible exports: iconNameToPathsRecordKey)

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/index.js 16:0-90
export 'IconSvgPaths20' (reexported as 'IconSvgPaths20') was not found in './iconSvgPaths' (possible exports: iconNameToPathsRecordKey)

After your fixes, new versions released and clean install with them, still same error:
```
ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconCodepoints.js 16:0-84
Module not found: Error: Can't resolve './generated/16px/blueprint-icons-16' in '{workspacePath}/node_modules/@blueprintjs/icons/lib/esm'
 @ ./node_modules/@blueprintjs/icons/lib/esm/index.js 17:0-72 17:0-72 17:0-72
 @ ./node_modules/@blueprintjs/core/lib/esm/components/icon/icon.js 19:0-94 56:63-77 56:80-94 57:35-59
 @ ./node_modules/@blueprintjs/core/lib/esm/components/index.js 47:0-28 47:0-28
 @ ./node_modules/@blueprintjs/core/lib/esm/index.js 18:0-29 18:0-29
 @ ./src/index.tsx 11:0-52 17:41-56

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconNames.js 19:0-72
Module not found: Error: Can't resolve './generated/16px/blueprint-icons-16' in '{workspacePath}/node_modules/@blueprintjs/icons/lib/esm'
 @ ./node_modules/@blueprintjs/icons/lib/esm/index.js 18:0-40 18:0-40
 @ ./node_modules/@blueprintjs/core/lib/esm/components/icon/icon.js 19:0-94 56:63-77 56:80-94 57:35-59
 @ ./node_modules/@blueprintjs/core/lib/esm/components/index.js 47:0-28 47:0-28
 @ ./node_modules/@blueprintjs/core/lib/esm/index.js 18:0-29 18:0-29
 @ ./src/index.tsx 11:0-52 17:41-56

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconSvgPaths.js 17:0-57
Module not found: Error: Can't resolve './generated/16px/paths' in '{workspacePath}/node_modules/@blueprintjs/icons/lib/esm'
 @ ./node_modules/@blueprintjs/icons/lib/esm/index.js 16:0-90 16:0-90 16:0-90 16:0-90
 @ ./node_modules/@blueprintjs/core/lib/esm/components/icon/icon.js 19:0-94 56:63-77 56:80-94 57:35-59
 @ ./node_modules/@blueprintjs/core/lib/esm/components/index.js 47:0-28 47:0-28
 @ ./node_modules/@blueprintjs/core/lib/esm/index.js 18:0-29 18:0-29
 @ ./src/index.tsx 11:0-52 17:41-56

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconSvgPaths.js 18:0-57
Module not found: Error: Can't resolve './generated/20px/paths' in '{workspacePath}/node_modules/@blueprintjs/icons/lib/esm'
 @ ./node_modules/@blueprintjs/icons/lib/esm/index.js 16:0-90 16:0-90 16:0-90 16:0-90
 @ ./node_modules/@blueprintjs/core/lib/esm/components/icon/icon.js 19:0-94 56:63-77 56:80-94 57:35-59
 @ ./node_modules/@blueprintjs/core/lib/esm/components/index.js 47:0-28 47:0-28
 @ ./node_modules/@blueprintjs/core/lib/esm/index.js 18:0-29 18:0-29
 @ ./src/index.tsx 11:0-52 17:41-56

webpack compiled with 4 errors
```
packages:
```
"@blueprintjs/core": "4.11.1"
"@blueprintjs/icons": "4.6.2"
"@blueprintjs/popover2": "1.7.1"
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -70,7 +70,10 @@ jobs:
       - run: yarn compile
       - persist_to_workspace:
           root: "."
-          paths: [packages/*/lib, packages/*/src/generated]
+          paths:
+            - packages/*/lib
+            - packages/*/src/generated
+            - packages/icons/src/generated-icons
 
   format-check:
     docker: *docker-node-image

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -78,5 +78,14 @@ module.exports = {
             },
         },
     ],
-    ignorePatterns: ["node_modules", "dist", "lib", "fixtures", "coverage", "__snapshots__", "generated"],
+    ignorePatterns: [
+        "node_modules",
+        "dist",
+        "lib",
+        "fixtures",
+        "coverage",
+        "__snapshots__",
+        "generated",
+        "generated-icons",
+    ],
 };

--- a/.gitignore
+++ b/.gitignore
@@ -18,11 +18,12 @@ yarn-error.log
 .sass-cache/
 
 # Blueprint
-build
-dist
-lib
-generated
-coverage
+build/
+dist/
+lib/
+generated/
+generated-icons/
+coverage/
 /docs
 docs-static/**/*.d.ts
 

--- a/.prettierignore
+++ b/.prettierignore
@@ -1,5 +1,6 @@
 node_modules/
 generated/
+generated-icons/
 lib/
 dist/
 /site/

--- a/packages/docs-data/package.json
+++ b/packages/docs-data/package.json
@@ -5,7 +5,7 @@
     "types": "src/index.d.ts",
     "private": true,
     "scripts": {
-        "clean": "rm -rf src/generated/*",
+        "clean": "rm -rf src/generated-icons/*",
         "compile": "node ./compile-docs-data.mjs",
         "test": "exit 0"
     },

--- a/packages/icons/package.json
+++ b/packages/icons/package.json
@@ -12,7 +12,7 @@
         "**/*.css"
     ],
     "scripts": {
-        "clean": "rm -rf dist/* && rm -rf lib/**/* && rm -rf src/generated/*",
+        "clean": "rm -rf dist/* && rm -rf lib/**/* && rm -rf src/generated-icons/*",
         "compile": "npm-run-all -s \"generate-icon-src\" -p \"compile:*\" -p \"copy:*\"",
         "compile:esm": "tsc -p ./src",
         "compile:cjs": "tsc -p ./src -m commonjs --outDir lib/cjs",

--- a/packages/icons/scripts/common.mjs
+++ b/packages/icons/scripts/common.mjs
@@ -21,7 +21,7 @@ import { fileURLToPath } from "node:url";
 
 export const scriptsDir = fileURLToPath(new URL(".", import.meta.url));
 export const iconResourcesDir = resolve(scriptsDir, "../../../resources/icons");
-export const generatedSrcDir = resolve(scriptsDir, "../src/generated");
+export const generatedSrcDir = resolve(scriptsDir, "../src/generated-icons");
 export const NS = "bp4";
 
 /**

--- a/packages/icons/scripts/copy-fonts.sh
+++ b/packages/icons/scripts/copy-fonts.sh
@@ -4,7 +4,7 @@
 # Usage: copy-fonts
 #   set INPUT / OUTPUT env varibles to change directories
 
-INPUT="${INPUT:-src/generated}"
+INPUT="${INPUT:-src/generated-icons}"
 OUTPUT="${OUTPUT:-lib/css}"
 
 mkdir -p "$OUTPUT"

--- a/packages/icons/scripts/copy-scss.sh
+++ b/packages/icons/scripts/copy-scss.sh
@@ -8,7 +8,7 @@ INPUT="${INPUT:-src}"
 OUTPUT="${OUTPUT:-lib/scss}"
 
 mkdir -p "$OUTPUT"
-cp "$INPUT/generated/16px/_icon-variables.scss" "$OUTPUT/blueprint-icons-16.scss"
-cp "$INPUT/generated/20px/_icon-variables.scss" "$OUTPUT/blueprint-icons-20.scss"
+cp "$INPUT/generated-icons/16px/_icon-variables.scss" "$OUTPUT/blueprint-icons-16.scss"
+cp "$INPUT/generated-icons/20px/_icon-variables.scss" "$OUTPUT/blueprint-icons-20.scss"
 
 cp "$INPUT/templates/_lib_variables.scss" "$OUTPUT/variables.scss"

--- a/packages/icons/scripts/generate-icon-paths.mjs
+++ b/packages/icons/scripts/generate-icon-paths.mjs
@@ -16,7 +16,7 @@
 /**
  * @fileoverview Generates SVG paths used in <Icon> React components
  *
- * Important: we expect ../src/generated/ to contain SVG definitions of all the icons already before this script runs.
+ * Important: we expect ../src/generated-icons/ to contain SVG definitions of all the icons already before this script runs.
  */
 
 // @ts-check

--- a/packages/icons/src/blueprint-icons.scss
+++ b/packages/icons/src/blueprint-icons.scss
@@ -3,5 +3,5 @@ Copyright 2017-present Palantir Technologies, Inc. All rights reserved.
 Licensed under the Apache License, Version 2.0.
 */
 
-@import "generated/16px/blueprint-icons-16";
-@import "generated/20px/blueprint-icons-20";
+@import "generated-icons/16px/blueprint-icons-16";
+@import "generated-icons/20px/blueprint-icons-20";

--- a/packages/icons/src/iconCodepoints.ts
+++ b/packages/icons/src/iconCodepoints.ts
@@ -15,7 +15,7 @@
  */
 
 // The two icon sets are identical aside from SVG paths, so we only need to import info for the 16px set
-import { BLUEPRINT_ICONS_16_CODEPOINTS } from "./generated/16px/blueprint-icons-16";
+import { BLUEPRINT_ICONS_16_CODEPOINTS } from "./generated-icons/16px/blueprint-icons-16";
 import type { IconName } from "./iconNames";
 
 /**

--- a/packages/icons/src/iconNames.ts
+++ b/packages/icons/src/iconNames.ts
@@ -19,7 +19,7 @@
 import { pascalCase, snakeCase } from "change-case";
 
 // The two icon sets are identical aside from SVG paths, so we only need to import info for the 16px set
-import { BlueprintIcons_16, BlueprintIcons_16Id as IconName } from "./generated/16px/blueprint-icons-16";
+import { BlueprintIcons_16, BlueprintIcons_16Id as IconName } from "./generated-icons/16px/blueprint-icons-16";
 import type { PascalCase, ScreamingSnakeCase } from "./type-utils";
 
 export type { IconName };

--- a/packages/icons/src/iconSvgPaths.ts
+++ b/packages/icons/src/iconSvgPaths.ts
@@ -16,8 +16,8 @@
 
 import { pascalCase } from "change-case";
 
-import * as IconSvgPaths16 from "./generated/16px/paths";
-import * as IconSvgPaths20 from "./generated/20px/paths";
+import * as IconSvgPaths16 from "./generated-icons/16px/paths";
+import * as IconSvgPaths20 from "./generated-icons/20px/paths";
 import type { IconName } from "./iconNames";
 import type { PascalCase } from "./type-utils";
 

--- a/scripts/get-publishable-paths-from-semver-tags.mjs
+++ b/scripts/get-publishable-paths-from-semver-tags.mjs
@@ -46,7 +46,7 @@ exec(`git tag --points-at ${commitish}`, (err, stdout) => {
             const packagePath = join("packages", unscopedName);
             return {
                 // This will throw if the package name isn't also the path, which is desirable.
-                packageJson: loadJsonFile(packagePath, "package.json"),
+                packageJson: loadJsonFile("..", packagePath, "package.json"),
                 path: packagePath,
             };
         })

